### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by butonic**
@LukasReschke this uses upstream with unnecessary files left out. As a result no longer uses ZendXml.
cc @DeepDiver1975

### CODE REVIEW

File: 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Docx.php [hunk 1]
```diff
@@ -62,12 +61,18 @@ private function __construct($fileName, $storeContent)
...
+        $relations = simplexml_load_string($relationsXml);
+
+        // Restore entity loader state
+        libxml_disable_entity_loader($loadEntities);
 
         foreach($relations->Relationship as $rel) {
             if ($rel ["Type"] == AbstractOpenXML::SCHEMA_OFFICEDOCUMENT) {
                 // Found office document! Read in contents...
-                $contents = XMLSecurity::scan($package->getFromName(
+                $contents = simplexml_load_string($package->getFromName(
```
Comment (LukasReschke) [inline 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Docx.php]: We seem to have this forgotten in the upstream patch then…?

File: 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Pptx.php [hunk 2]
```diff
@@ -82,25 +81,31 @@ private function __construct($fileName, $storeContent)
...
+        $relations = simplexml_load_string($relationsXml);
+
+        // Restore entity loader state
+        libxml_disable_entity_loader($loadEntities);
 
         foreach ($relations->Relationship as $rel) {
             if ($rel["Type"] == AbstractOpenXML::SCHEMA_OFFICEDOCUMENT) {
                 // Found office document! Search for slides...
-                $slideRelations = XmlSecurity::scan($package->getFromName( $this->absoluteZipPath(dirname($rel["Target"]) . "/_rels/" . basename($rel["Target"]) . ".rels")) );
+                $slideRelations = simplexml_load_string($package->getFromName( $this->absoluteZipPath(dirname($rel["Target"]) . "/_rels/" . basename($rel["Target"]) . ".rels")) );
```
Comment (LukasReschke) [inline 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Pptx.php]: Needs adjustment as well as the changed lines below then.

File: 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Xlsx.php [hunk 3]
```diff
@@ -91,18 +90,24 @@ private function __construct($fileName, $storeContent)
...
+        $relations = simplexml_load_string($relationsXml);
+
+        // Restore entity loader state
+        libxml_disable_entity_loader($loadEntities);
 
         foreach ($relations->Relationship as $rel) {
             if ($rel["Type"] == AbstractOpenXML::SCHEMA_OFFICEDOCUMENT) {
                 // Found office document! Read relations for workbook...
-                $workbookRelations = XmlSecurity::scan($package->getFromName($this->absoluteZipPath(dirname($rel["Target"]) . "/_rels/" . basename($rel["Target"]) . ".rels")));
+                $workbookRelations = simplexml_load_string($package->getFromName( $this->absoluteZipPath(dirname($rel["Target"]) . "/_rels/" . basename($rel["Target"]) . ".rels")) );
```
Comment (LukasReschke) [inline 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Xlsx.php]: Same

File: 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Xlsx.php [hunk 4]
```diff
@@ -116,7 +121,7 @@ private function __construct($fileName, $storeContent)
                 // Loop relations for workbook and extract worksheets...
                 foreach ($workbookRelations->Relationship as $workbookRelation) {
                     if ($workbookRelation["Type"] == self::SCHEMA_WORKSHEETRELATION) {
-                        $worksheets[ str_replace( 'rId', '', (string)$workbookRelation["Id"]) ] = XmlSecurity::scan(
+                        $worksheets[ str_replace( 'rId', '', (string)$workbookRelation["Id"]) ] = simplexml_load_string(
```
Comment (LukasReschke) [inline 3rdparty/zendframework/zendsearch/library/ZendSearch/Lucene/Document/Xlsx.php]: Same

**Comment by LukasReschke** [comment]
+1 for the TCPDF thing. Regarding ZF Search we either missed to replace some things or something mixed up.

So either we patch those as well or we keep the current ZFSearch version (maybe with my merged patch upstream applied?)

**Comment by LukasReschke** [comment]
Any chance we can bump TCPDF for 8.0? This is important. – THX!

I'd stick with our custom ZendSearch version then for now until I found some time to sort the upstream mess out.

**Comment by butonic** [comment]
superseded by https://github.com/owncloud/search_lucene/pull/54, cllosing

**Pull request closed without merging**